scim2-sdk-server/src/main/java/com/unboundid/scim2/server/utils/SchemaChecker.java

```diff
@@ -634,9 +631,8 @@ private void checkPartialResource(
       final boolean isPartialAdd)
           throws ScimException
   {
-
-    Iterator<Map.Entry<String, JsonNode>> i = objectNode.fields();
-    while (i.hasNext())
+    Iterator<Map.Entry<String, JsonNode>> i;
+    for (i = objectNode.properties().iterator(); i.hasNext();)
```

Comment:
These usages need the iterator, since they are removing nodes with `i.remove()`.